### PR TITLE
Introduce chrome.braveTheme api

### PR DESCRIPTION
**Pull request by simonhong**
With this new api, shields, rewards and many other webui can get brave
theme information.

Issue brave/brave-browser#793

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
`yarn test brave_browser_tests --filter=BraveThemeAPIBrowserTest*`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by simonhong** [comment]
Will remove `WIP` after adding `onBraveThemeTypeChanged` event.

**Comment by simonhong** [comment]
event is added . Will remove WIP after adding test code for event.

**Comment by simonhong** [comment]
Kindly ping..

File: common/extensions/api/brave_theme.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        "parameters": [
+          {
+            "name": "callback",
+            "type": "function",
+            "description": "Function called when current theme type is fetched",
+            "parameters": [
+              {
+                "name": "type",
+                "type": "string",
+                "description": "current theme type(ex, Dark or Light)"
```
Comment (bbondy) [inline common/extensions/api/brave_theme.json]: I guess we'd need to also expose the channel to be able to know if it's actually dark or light if the value is default right?   I think Default can be another value, so pls indicate here and above for that.

**Comment by petemill** [comment]
Apologies for the delay - I was out sick. On the topic of @bbondy 's comment, I'd go further and suggest that this JS API should **not** expose `GetUserPreferredBraveThemeType` (which could be `default, dark, light`) but should instead expose `GetActiveBraveThemeType` which can only be `dark, light`.
Shouldn't we should keep all the logic about how `default` is interpreted in one place? It will also not matter in almost all circumstances whether the theme is chosen directly from the user or is the default.

In fact, we can use `GetActiveBraveThemeType` and still only keep 1 API as even the settings page doesn't really need to know about `default` or provide the 'default' option to the user - it should just provide `dark` and `light` and it will already be defaulted because `GetActiveBraveThemeType` will translate `default` to the appropriate value.

**Comment by petemill** [comment]
(another reason not to introduce parsing `Default` in another layer is that the result of that logic is subject to change, e.g. if / when our default changes to follow OS-set light / dark value.

**Comment by simonhong** [comment]
I agree with @petemill. Default isn't another different theme option.
It's just translated to existing another option. @bbondy WDYT?

**Comment by petemill** [comment]
Received 👍 from design team on removing Default option for theme on Settings page so hopefully this is a straightforward call swap

**Comment by simonhong** [comment]
GetUserPreferredBraveThemeType() is replaced with GetActiveBraveThemeType() in theme api.
Also `default` option is removed from brave theme option list.

**Comment by bbondy** [comment]
>  I'd go further and suggest that this JS API should not expose GetUserPreferredBraveThemeType (which could be default, dark, light) but should instead expose GetActiveBraveThemeType which can only be dark, light.

I considered that too but wondered if it'll ever be useful for us to know if the user has changed from what the default is or not. I'm ok with always doing light or dark though.

**Comment by simonhong** [comment]
@petemill I think this PR is ready to merge :)

**Comment by petemill** [comment]
@simonhong I'm getting the following build error:

```
ERROR at //chrome/chrome_repack_locales.gni:5:1: Can't load input file.
import("//brave/brave_repack_locales.gni")
^----------------------------------------
Unable to load:
  /Users/xyz/Development/Brave/brave-browser/src/brave/brave_repack_locales.gni
I also checked in the secondary tree for:
  /Users/xyz/Development/Brave/brave-browser/src/build/secondary/brave/brave_repack_locales.gni
See //chrome/chrome_paks.gni:9:1: whence it was imported.
import("chrome_repack_locales.gni")
^---------------------------------
See //chrome/BUILD.gn:16:1: whence it was imported.
import("//chrome/chrome_paks.gni")
^--------------------------------
See //BUILD.gn:101:7: which caused the file to be included.
      "//chrome",
```

**Comment by bbondy** [comment]
You likely have a patch applied unrelated to this commit that shouldn't exist for this branch. You can rebase to master or re-init.

**Comment by simonhong** [comment]
@petemill Rebased this PR on latest master now.

**Comment by simonhong** [comment]
@petemill Thanks! 👍

**Comment by simonhong** [comment]
@bbondy For this PR, https://github.com/brave/brave-browser/issues/2348 is created and closed.
I think https://github.com/brave/brave-browser/issues/793 could be closed after our webui pages are aware about dark/light theme.

**Pull request merged**